crates/red_knot_python_semantic/src/types.rs

```diff
@@ -328,8 +328,7 @@ impl<'db> Type<'db> {
         match self {
             Type::Function(function_type) => function_type.returns(db).or(Some(Type::Unknown)),
 
-            // TODO: handle class constructors
```

Comment:
Sooo... I think we should keep _some_ TODO comment here, because there are unfortunately some edge cases we'll need to handle in due course:
- Classes with `__new__` methods that are annotated as not returning an instance of the class
- Classes where the metaclass has a __call__ method with an unusual return annotation

These are edge cases that we _definitely_ don't need to worry about now, but a TODO comment would be great IMO